### DISCUSSION TITLE
Can the Draw2D functions be used in conjunction with a camera in a sub viewport?

### DISCUSSION THREAD

**Discussion by ALapidis** [DmitriySalnikov/godot_debug_draw_3d#30] [General]
I'm working on a main screen plugin with a sub viewport/camera filling the viewport. The DebugDraw2D doesn't function here and was curious if there was a way to enable the 2D elements to draw in the sub-viewport or even in the plugin's viewport window. 3D elements work fine, but the 2D must be specific to the 3D tabs camera?

**Comment by DmitriySalnikov** [comment]
Do you mean a behavior similar to what happens when you press <kbd>Shift</kbd> in the demo scene?
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/b205b614-9088-4427-af73-27c287b1f5a7)
```gdscript
DebugDraw2D.custom_canvas = $CustomCanvas
```

But at the moment, you can specify only one canvas. In the future, I will consider adding support for multiple canvases.

**Reply by ALapidis** [reply]
Hmm, I see what you mean. Shift moves the Debug2D to the CustomCanvas and Displays the 3D elements in the sub viewport. But can the Debug2D elements also be rendered in the custom viewport along with the Debug3D examples?

**Reply by DmitriySalnikov** [reply]
Well, it even seems to work in a separate window.
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/fd9e0a51-412f-49fd-9d54-f6ed1310d5f7)

At the moment, you can select any viewport, but only one. More precisely, only the `Control` is selected on which everything will be drawn.

**Reply by ALapidis** [reply]
Thank you for the help I'll keep at it, I'm sure it's something I am missing on my end. And thank you so much for this incredible add-on, its been a huge time saver.  😄

**Reply by ALapidis** [reply]
Got it working by assigning the canvas as you suggested thank you!

**Reply by DmitriySalnikov** [reply]
> At the moment, you can select any viewport, but only one. More precisely, only the Control

I mean this structure:
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/391d82b3-9c38-4a6a-8b11-f7f5fb02ddb8)

> And thank you so much for this incredible add-on, its been a huge time saver. 😄

I'm glad to hear it. I hope to finish a big update soon
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/d4efc2eb-190e-4395-811c-be3a5460cd2f)

---

> Got it working by assigning the canvas as you suggested thank you!

You are welcome!

**Reply by ALapidis** [reply]
Yeah, It was my own fault, I was assigning it "behind" the sub-viewport. /headslap.
